src/API/index.js

```diff
@@ -1,87 +1,121 @@
-  let errorText = '';
-  if (err.response) {
-    console.log('Error Status', err.response.status);
-    console.log('Error Downloading', err.response.data);
-    errorText = err.response.data;
-  } else if (err.request) {
-    console.log('No response was received', err.request);
-    errorText = 'There was no response from the server.';
-  } else {
-    console.log('Unknown Error', err.message);
-    errorText = err.message;
-  }
-  return errorText;
-  /* eslint-enable no-console */
+const protocol = 'http';
+const host = process.env.ROBOKACHE_HOST || 'lvh.me';
+const port = 8080;
+
+/**
+ * URL Maker
+ * @param {string} ext extension to append to url
+ */
+const base_url = `${protocol}://${host}:${port}/api/`;
+
+function handleAxiosError(error) {
+	let errorResponse;
+	if (error.response) {
+		errorResponse = error.response.data;
+		errorResponse.status = "error";
+	} else {
+		// This either means the server is unreachable or there was
+		// some error setting up the request object
+		errorResponse = {
+			message: "An unknown error occured",
+			status: "error"
+		}
+	}
+	return {data: errorResponse};
 }
 
-const routes = {
-  getUser: () => new Promise((resolve, reject) => {
-    axios.get('/api/user')
-      .then((res) => {
-        console.log('user res', res);
-        resolve(res.data);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-  setUser: (token, username) => axios.post('/api/user', { token, username }),
-  getAllQuestions: () => new Promise((resolve, reject) => {
-    axios.get('/api/questions')
-      .then((res) => {
-        console.log('Questions', res);
-        resolve(res);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-  createQuestion: (question, token) => new Promise((resolve, reject) => {
-    const config = {
-      method: 'get',
-      url: '/api/questions',
-      headers: {},
-    };
-    if (token) {
-      config.headers.authorization = `Bearer ${token}`;
-    }
-    axios.request({
-      method: 'post',
-      url: '/api/questions',
-      data: question,
-    })
-      .then((res) => {
-        resolve(res);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-  getQuestions: (token) => new Promise((resolve, reject) => {
-    const config = {
-      method: 'get',
-      url: '/api/questions',
-      headers: {},
-    };
+// Base request method for all endpoints
+async function baseRequest(path, method, body, token) {
+	let config = {
+		url: base_url + path,
+		method: method,
+		data: body,
+		withCredentials: true,
+		headers: {
+			'Accept': 'application/json',
+			'Content-Type': 'application/json',
+		},
+	};
+	if (token) {
+		config.headers.Authorization = `Bearer ${token}`;
+	}
+
+	let response = await axios(config).catch(handleAxiosError);
```

Comment:
I'll admit, I'm used to writing promises with then/catch. But we can switch over to async/await. But I would suggest we stick with one idiom and not mix. This should probably be in a try/catch block?